src/tbb/task.cpp

```diff
@@ -236,9 +237,11 @@ d1::wait_tree_vertex_interface* get_thread_reference_vertex(d1::wait_tree_vertex
         if (reference_map.size() > max_reference_vertex_map_size) {
             // TODO: Research the possibility of using better approach for a clean-up
             for (auto it = reference_map.begin(); it != reference_map.end();) {
-                if (it->second->get_num_child() == 0) {
-                    it->second->~reference_vertex();
-                    cache_aligned_deallocate(it->second);
+                if (it->second->get_num_children() == 0) {
+                    auto& node = it->second;
+                    __TBB_ASSERT(!(node->m_ref_count.load(std::memory_order_relaxed) & node->m_orphaned_bit),
+                        "the orphaned bit cannot be set");
```

Comment:
Since `get_num_children` was fixed to mask out the orphaned bit, I think we can remove the assertion here. Now it seems like we are accessing private field (with a friend) just to do an assertion.

Reply:
I think even a little bit silly assert is still good to have in case if we somehow mess up with `get_num_children` in future.

Reply:
But we can move the assertion out of `if`, can't we? In which case it would make more sense.

Reply:
Moved.